File: src/099. Largest exponential/099.rb
max = 0
max_line_no = 0

File.open("p099_base_exp.txt") do |file|
  line_no = 1
  while line = file.gets
    md = line.match(/^(\d+),(\d+)$/)
	base, exp = md[1].to_i, md[2].to_i
	current = exp * Math.log(base)
	if current > max
	  max = current
	  max_line_no = line_no
	end
	line_no += 1
  end
end

puts max_line_no

File: src/099. Largest exponential/099.c
#include <stdio.h>
#include <string.h>
#include <math.h>

#define LINE_LEN 100

int main(void) {
    FILE *f = fopen("p099_base_exp.txt", "r");
    char line[LINE_LEN];
    double max_power = 0;
    int max_line_no = 0;
    int line_no = 0;
    while (fgets(line, sizeof(line), f)) {
        line_no += 1;
        int base, exp;
        sscanf(line, "%d,%d", &base, &exp);
        double current_power = exp * log((double)base);
        if (current_power > max_power) {
            max_power = current_power;
            max_line_no = line_no;
        }
    }
    printf("%d\n", max_line_no);
}
